### PR TITLE
Update license headers

### PR DESCRIPTION
**Pull request by systay**
This PR does two things:
 - it includes a little go script that checks *.go files for the correct license header. This script is run as a pre-commit check.
 - The script can also update go files to use the correct license header. The second commit updates the license headers on a lot of source files to match the configured one.

### CODE REVIEW

File: misc/git/hooks/license-check [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+# Copyright 2017 Google Inc.
```
Comment (systay) [inline misc/git/hooks/license-check]: OMG THE IRONY!

**Comment by deepthi** [comment]
I find that it is good practice to create PRs with a descriptive title and to fill in the PR description with a summary of the changes. This is helpful to people who are searching for specific PRs or just browsing through the repo.

**Comment by systay** [comment]
@deepthi You are absolutely right. I was in a hurry when I created it, and that shows. Updated now.

**Comment by deepthi** [comment]
what happens to https://github.com/vitessio/vitess/blob/master/go/vt/vterrors/LICENSE?

**Comment by systay** [comment]
@deepthi Not sure what a better solution looks like. This last commit restores the code to the state it was in earlier at least.

**Comment by tanjunchen** [comment]
@sougou  @systay 
Some thoughts:
* About the license check year, I find [Kubernetes](https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/gci/audit_policy_test.go) has  different year in different files,but  [containerd](https://github.com/containerd/containerd/blob/master/gc/gc.go) has not year. The Lisence Header in  .go files,.java files,.py files etc. has two format in the vitess repo. The owner of vitess repo need to talk about it .  I think  we need to unite the format and  my view is **Copyright  The Vitess Authors.**
* I agree the point that separating PR about license,the issue #5279 is great. we can clean up them in directory format such as go/,java/,py/. The Check  Program maybe can improved,excluding some files and containing DO NOT EDIT. files that generated from third party. [refrence](https://github.com/kubernetes/kubernetes/blob/master/hack/boilerplate/boilerplate.py) can  give a example.
* The change is huge. Maybe Squash them is good. 
Thanks

**Comment by systay** [comment]
Seems like concensus does not think that merging this is a good idea. Closing for now. Please plunder any good things from it as you see fit.

**Pull request closed without merging**